action-server/covidflow/actions/assessment_form.py

```diff
@@ -1,158 +1,122 @@
-from typing import Any, Dict, List, Optional, Text, Union
+from typing import Any, Dict, List, Text
 
-from rasa_sdk import Tracker
+import structlog
+from rasa_sdk import Action, Tracker
 from rasa_sdk.events import EventType, SlotSet
 from rasa_sdk.executor import CollectingDispatcher
-from rasa_sdk.forms import FormAction
+from rasa_sdk.forms import REQUESTED_SLOT
 
 from covidflow.constants import (
-    CONTACT_SLOT,
-    HAS_CONTACT_RISK_SLOT,
+    AGE_OVER_65_SLOT,
     HAS_COUGH_SLOT,
     HAS_FEVER_SLOT,
-    LIVES_ALONE_SLOT,
+    MODERATE_SYMPTOMS_SLOT,
+    PROVINCE_SLOT,
+    PROVINCES,
+    PROVINCIAL_811_SLOT,
+    SELF_ASSESS_DONE_SLOT,
     SEVERE_SYMPTOMS_SLOT,
-    TRAVEL_SLOT,
+    SKIP_SLOT_PLACEHOLDER,
+    SYMPTOMS_SLOT,
+    Symptoms,
 )
 
-from .assessment_common import AssessmentCommon
-from .lib.form_helper import (
-    request_next_slot,
-    validate_boolean_slot,
-    yes_no_nlu_mapping,
-)
 from .lib.log_util import bind_logger
+from .lib.provincial_811 import get_provincial_811
 
-FORM_NAME = "assessment_form"
+logger = structlog.get_logger()
 
+FORM_NAME = "assessment_form"
+VALIDATE_ACTION_NAME = f"validate_{FORM_NAME}"
+ASK_PROVINCE_ACTION_NAME = "action_ask_province_code"
 
-class AssessmentForm(FormAction, AssessmentCommon):
-    def name(self) -> Text:
+ORDERED_FORM_SLOTS = [
+    SEVERE_SYMPTOMS_SLOT,
+    PROVINCE_SLOT,
+    AGE_OVER_65_SLOT,
+    HAS_FEVER_SLOT,
+    MODERATE_SYMPTOMS_SLOT,
+    HAS_COUGH_SLOT,
+]
 
-        return FORM_NAME
 
-    async def run(
-        self, dispatcher, tracker, domain,
-    ):
-        bind_logger(tracker)
-        return await super().run(dispatcher, tracker, domain)
+class ActionAskProvinceCode(Action):
+    def name(self) -> Text:
+        return ASK_PROVINCE_ACTION_NAME
 
-    ## override to play initial message
-    async def _activate_if_required(
+    def run(
```

Comment:
I'm going to be a bit annoying with this. I know it's not forced to put this tag but the interface is `async` and I think it's important to keep it 'documented' since this function will always be run asynchronously within Rasa.
```suggestion
    async def run(
```

Reply:
Totally agree. I'll try to think about it when I create or modify actions and I'll create an issue to change the remaining ones one day.

Reply:
Changed my mind. There are complications to changing only some of them (with tests and helpers), so I'll change them all in one PR that I'll do later can't do only these now.